src/torrent.cpp

```diff
@@ -5655,6 +5650,22 @@ namespace {
 
 	namespace
 	{
+		std::vector<boost::uint8_t> fix_priorities(std::vector<int> const& input, file_storage const& fs)
```

Comment:
factoring out this logic into a pure function is clearly a good thing. However, I'm a bit worried about merging this patch (correctly) into master. I'm a bit tempted to factor this function out in master instead of here, to keep this patch smaller

Reply:
I suppose this is required, because we need to convert `std::vector<int>` to `std::vector<std::uint8_t>` in order to make the disk thread call

Reply:
Yes, also to reassign invalid priorities with correct values.

Reply:
Hmm, will it make patch's size much smaller? However it's not a problem to move this code back to the `torrent::prioritize_files`.

Reply:
I left some comments on it. Please take a look.